File: Problem 10.py
'''
The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

Find the sum of all the primes below two million.
'''

def isPrime(n):
	if (n==0 or n==1):
		return 0
	if (n==2):
		return 1
	for x in range(2,n):
		if(n%x==0):
			return 0
		else:
			pass
	return 1

def sum_primes_below(n):
    i = 0
    summ = 0
    while (i < n):
        if (isPrime(i)==1):
            summ += i
        i = i + 1
    print(summ)
    return(summ)

sum_primes_below(2000000)

File: Problem 10.c
#include <stdio.h>

int isPrime(int n){
	if (n == 0 | n == 1)
		return 0;
	if (n == 2)
		return 1;
	for (int i = 2; i < n; i++) {
		if (n%i == 0)
			return 0;
	}
	return 1;
}

unsigned long long int sum_primes_below(unsigned int n) {
	unsigned int i = 0;
	unsigned long long int summ = 0;
	while (i<n){
		if (isPrime(i) == 1){
			summ += i;
			//printf("%u\t%llu\n", i, summ);
		}
		i++;
	}
	printf("%llu\n", summ);
	return summ;
}

void main() {
	sum_primes_below(2000000);
	getchar();
}

//output 142913828922
